### PR TITLE
Preview Movie Rec Tutorial

### PR DESCRIPTION
**Pull request by briacht**
## Summary

Movie Recommender Tutorial

[Docs Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/movie-recommmendation?branch=pr-en-us-11329)

### CODE REVIEW

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+| 1 | 1 | 4 | userKey1 | movieKey1 |
+| 1 | 3 | 4 | userKey1 | movieKey2 |
+| 1 | 6 | 4 | userKey1 | movieKey3 |
+
+
+Choose the machine learning algorithm and append it to the data transformation definitions by adding the following as the next line of code in `BuildAndTrainModel()`:
+
+[!code-csharp[AddAlgorithm](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#AddAlgorithm "Add the training algorithm with options")]
+
+The [MatrixFactorizationTrainer](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization%28Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options%29) as is your recommendation training algorithm.  [Matrix Factorization](https://en.wikipedia.org/wiki/Matrix_factorization_(recommender_systems)) is a common approach to recommendation when you have data on how users have rated products in the past, which is the case for the datasets in this tutorial. There are other recommendation algorithms for when you have different data available (see the [Recommended Resources](#recommended-resources) section below to learn more).
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: Small typo here: need to remove the "as" from "as is your recommendation training algorithm"

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+The [LoadFromTextFile()](xref:Microsoft.ML.TextLoaderSaverCatalog.LoadFromTextFile%60%601%28Microsoft.ML.DataOperationsCatalog,System.String,System.Char,System.Boolean,System.Boolean,System.Boolean,System.Boolean%29) defines the data schema and reads in the file. It takes in the data path variables and returns an `IDataView`. In this case, you provide the path for your `Test` and `Train` files and indicate both the text file header (so it can use the column names properly) and the comma character data separator (the default separator is a tab).
+
+Add the following as the next two lines of code in the `Main()` method to call your `LoadData()` method and return the `Train` and `Test` data:
+
+[!code-csharp[LoadDataMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#LoadDataMain "Add LoadData method to Main")]
+
+
+## Build and train your model
+
+There are three major concepts in ML.NET: [Data](../basic-concepts-model-training-in-mldotnet.md#data), [Transformers](../basic-concepts-model-training-in-mldotnet.md#transformer), and [Estimators](../basic-concepts-model-training-in-mldotnet.md#estimator).
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: I think we can improve this section, but we can do that post-publication.

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+In this case, the `Matrix Factorization` algorithm uses a method called "collaborative filtering", which assumes that if User 1 has the same opinion as User 2 on a certain issue, then User 1 is more likely to feel the same way as User 2 about a different issue.
+
+E.g. if User 1 and User 2 rate movies similarly, then User 2 is more likely to enjoy a movie that User 1 has watched and rated highly:
+
+|       | Incredibles 2 (2018) | The Avengers (2012) | Guardians of the Galaxy (2014) |
+| -------------:|-------------:| -----:|-----:|
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options) including:
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: Can we move this section into the "Improve your model" section. We might lose some users in the explanation at this point.

File: docs/machine-learning/tutorials/movie-recommmendation.md (hunk 1 above)
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: Also, need to update [Recommended Resources](#recommended-resources) --> [Resources](#resources)

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+
+In this output, there are 20 iterations. In each iteration, the measure of error decreases and converges closer and closer to 0.
+
+The `root of mean squared error` (RMS or RMSE) is frequently used to measure the differences between values predicted by a model and the values actually observed in a test dataset. Technically it's the square root of the average of the squares of the errors. You want your RMSE score to be as close to 1 as possible.
+
+`R Squared` is the variation percentage in the predicted values explained by your model. It's a value between 0 and 1, and the closer the value is to 0, the better the model is.
+
+Add the following as the next line of code in the `Main()` method to call your `EvaluateModel()` method:
+
+[!code-csharp[EvaluateModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModelMain "Add EvaluateModel method in Main")]
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: Do we need to move this up above the output sections? i.e. we won't see the output until this method is called.

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+
+You can then use the `Score`, or the predicted rating, to determine whether you want to recommend the movie with movieId 10 to user 6. The higher the `Score`, the higher the likelihood of a user liking a particular movie. In this case, let’s say that you recommend movies with a predicted rating of > 3.5.
+
+To print the results, add the following as the next lines of code in the `UseModelForSinglePrediction()` method:
+
+[!code-csharp[PrintResults](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#PrintResults "Print the recommendation prediction results")]
+
+Add the following as the next line of code in the `Main()` method to call your `UseModelForSinglePrediction()` method:
+
+[!code-csharp[UseModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#UseModelMain "Add UseModelForSinglePrediction method in Main")]
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: Would we want to show the output of this method?

**Review by natke — CHANGES_REQUESTED** [review]
Hi @briacht, I have made a few minor comments. Other than that, looks good!

File: docs/machine-learning/tutorials/movie-recommmendation.md (hunk 2 above)
Comment (briacht) [reply docs/machine-learning/tutorials/movie-recommmendation.md]: Agreed!

File: docs/machine-learning/tutorials/movie-recommmendation.md (hunk 3 above)
Comment (briacht) [reply docs/machine-learning/tutorials/movie-recommmendation.md]: I think we can, but there will be the Options/hyperparameters visible in the code snippet, so users might wonder about that. I moved it down below and just added a "learn more below" link, let me know what you think

File: docs/machine-learning/tutorials/movie-recommmendation.md [hunk 6]
```diff
@@ -0,0 +1,474 @@
...
+In this case, the `Matrix Factorization` algorithm uses a method called "collaborative filtering", which assumes that if User 1 has the same opinion as User 2 on a certain issue, then User 1 is more likely to feel the same way as User 2 about a different issue.
+
+For instance, if User 1 and User 2 rate movies similarly, then User 2 is more likely to enjoy a movie that User 1 has watched and rated highly:
+
+| | `Incredibles 2 (2018)` | `The Avengers (2012)` | `Guardians of the Galaxy (2014)` |
+| -------------:|-------------:| -----:|-----:|
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options, which you can read more about in the [Algorithm hyperparameters](#algorithm-hyperparameters) section below.
```
Comment (natke) [inline docs/machine-learning/tutorials/movie-recommmendation.md]: One tiny typo '['.

**Review by natke — APPROVED** [review]

**Pull request merged**